README.md

```diff
@@ -1,32 +1,122 @@
+
+Usually an application would want multiple controllers. However, it may not be efficient for every controller to have its own informer and therefore its own indexer. The SharedIndexerInformer from client-go can be used to add multiple controllers that all use the same indexers.
+
+There is a great visual of how a Kubernetes controller might work found in the [kubernetes/sample-controller repo](https://github.com/kubernetes/sample-controller/blob/master/docs/images/client-go-controller-interaction.jpeg).
+
+__Note:__ Informers have been simplified for the sake of clarity. If you are interested in exactly how they work it is recommended you read through the [client-go code](https://github.com/kubernetes/client-go).
+
+# Lasso's Value Proposition
+### Lasso Controller
+Lasso standardizes what a controller is with its controller package. Lasso introduces factories for managing caches, controllers, and clients within a project. This is imperative for rancher which manages many instances of many types.
+
+The lasso controller uses the client-go work queue. Lasso registers an event handler that processes the work queue. The work queue oriented handler makes it so only one event handler needs to be added for each Kubernetes type. Now registering a handler just adds it to the type's controller's list of handlers. Enqueueing an objects adds it to the work queue. Each object in the work queue will then be processed by workers running in goroutines. The controller's event handler will re-enqueue any object that causes one or more of the controller's handlers to return an error that is not of type `ErrIgnore`. This work queue can also have objects enqueued from other sources. This functionality is exposed through the controller's Enqueue function. This transforms controllers from only being reactive to Kubernetes resources and cache re-syncs, to also being invokable by code.
+
+### Lasso Shared Controller
+The sharedController type embeds the Controller interface to expose its controller's functionality. The sharedController also exposes the RegisterHandler function along with the Client function. RegisterHandler adds the passed handler to the controller's handler list and Client returns a client for the sharedController corresponding resource type; sharedController has a GVK field (Group Version Kind) which indicates what type it is for.
+
+The sharedController can create its underlying controller. It will only do so once a handler is registered to the sharedController.
+
+### Lasso Client
+Wraps Kubernetes client. Contains GVK fields for managing the client in a parent struct such as the sharedClientFactory. Exposes some additional functionality like configuring the User-Agent header for the underlying REST client.
+
+### Lasso Cache
+The lasso Cache is a light wrapper around a [SharedIndexInformer from client-go](https://pkg.go.dev/k8s.io/client-go/tools/cache#SharedIndexInformer).
```

Comment:
Why should I use it instead of SharedIndexInformer?

Reply:
Added.